File: lib/src/androidTest/java/com/pigdogbay/lib/utils/LetterSetTest.java
package com.pigdogbay.lib.utils;

import androidx.test.ext.junit.runners.AndroidJUnit4;

import org.junit.Test;
import org.junit.runner.RunWith;

import static org.hamcrest.Matchers.is;
import static org.junit.Assert.assertThat;

@RunWith(AndroidJUnit4.class)
public class LetterSetTest {

    @Test
    public void isAnagram1(){
        //cat+++
        LetterSet letterSet = new LetterSet("cat");
        assertThat(letterSet.isAnagram("chart",3),is(true));
        assertThat(letterSet.isAnagram("charts",3),is(true));
        assertThat(letterSet.isAnagram("charted",3),is(false));
    }
    @Test
    public void isAnagram2(){
        LetterSet letterSet = new LetterSet("");
        assertThat(letterSet.isAnagram("",0),is(true));
        assertThat(letterSet.isAnagram("cat",3),is(true));
        assertThat(letterSet.isAnagram("cats",3),is(false));
    }
    @Test
    public void isAnagram3(){
        LetterSet letterSet = new LetterSet("black");
        assertThat(letterSet.isAnagram("black",0),is(true));
        assertThat(letterSet.isAnagram("clack",0),is(false));
        assertThat(letterSet.isAnagram("clack",1),is(true));
    }

    /*
        Regression test for a bug, Letterset could not handle spaces
     */
    @Test
    public void isSubgram(){
        LetterSet letterSet = new LetterSet("words with friends");
        assertThat(letterSet.isSubgram("finds"),is(true));
    }
}

File: lib/src/test/java/com/pigdogbay/lib/utils/LetterSetTest.kt
package com.pigdogbay.lib.utils
import org.junit.Test

import org.junit.Assert.*

class LetterSetTest {
    @Test
    fun isAnagram1() {
        //cat+++
        val letterSet = LetterSet("cat")
        assertTrue(letterSet.isAnagram("chart", 3))
        assertTrue(letterSet.isAnagram("charts", 3))
        assertFalse(letterSet.isAnagram("charted", 3))
    }

    @Test
    fun isAnagram2() {
        val letterSet = LetterSet("")
        assertTrue(letterSet.isAnagram("", 0))
        assertTrue(letterSet.isAnagram("cat", 3))
        assertFalse(letterSet.isAnagram("cats", 3))
    }

    @Test
    fun isAnagram3() {
        val letterSet = LetterSet("black")
        assertTrue(letterSet.isAnagram("black", 0))
        assertFalse(letterSet.isAnagram("clack", 0))
        assertTrue(letterSet.isAnagram("clack", 1))
    }

    /*
        Regression test for a bug, Letterset could not handle spaces
     */
    @Test
    fun isSubgram() {
        val letterSet = LetterSet("words with friends")
        assertTrue(letterSet.isSubgram("finds"))
    }

    @Test
    fun isDistinct1() {
        val letterSet = LetterSet("spectrum")
        assertTrue(letterSet.isDistinct)
    }

    @Test
    fun isDistinct2() {
        val letterSet = LetterSet("electron")
        assertFalse(letterSet.isDistinct)
    }

    @Test
    fun isDistinct3() {
        val letterSet = LetterSet("")
        assertFalse(letterSet.isDistinct)
    }

}